### PR TITLE
Reduce iterations in TileEntities activation

### LINKED ISSUE #2780: TileEntityActivation Performance
**I am currently running**

- SpongeForge version: 3720
- Forge version: 2836
- Java version: 1.8_201
- Operating System: Windows Server 2016

- Plugins/Mods: FTBRevelations 3.0.1 + https://pastebin.com/raw/QVBKZm3A

**Issue Description**

Performance impact from tile activation when controlling tick rates of entities.

I am using tile activation to control tick rates. I have the file populated and many of the mods disabled by default but still experiencing a hit from tile activation. 

Spark: https://sparkprofiler.github.io/#ThhUaANL8X
Sponge global.conf: https://pastebin.com/raw/Cxd4Nf5P

I can also provide a nps report of the same behavior. I can also provide a test build with the server configuration, this map is less than 5 days old with 30 players average, not many tile entities present.

Thanks.

### PR DESCRIPTION
**Pull request by ImMorpheus**
Fix https://github.com/SpongePowered/SpongeForge/issues/2780

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/plugin/tileentityactivation/TileEntityActivation.java [hunk 1]
```diff
@@ -177,9 +176,13 @@ private static void activateChunkTileEntities(final EntityPlayer player, final C
                 }
 
                 final int bbActivationRange = ((ActivationCapability) tileEntity).activation$getActivationRange();
-                final int blockDistance = Math.round(tilePos.distance(playerPos));
-                if (blockDistance <= bbActivationRange) {
-                    ((ActivationCapability) tileEntity).activation$setActivatedTick(currentTick);
+                for (EntityPlayerMP player : players) {
+                    final Vector3i playerPos = VecHelper.toVector3i(player.getPosition());
+                    final int blockDistance = Math.round(tilePos.distance(playerPos));
```
Comment (dualspiral) [inline src/main/java/org/spongepowered/common/mixin/plugin/tileentityactivation/TileEntityActivation.java]: While we're looking at performance, I wonder if we should also look at making this dependent on distance squared. I realise that it makes some difference to getting `bbActivationRange` and `Math.round()` - but the activation range would then be `Math.pow(bbActivationRange - 0.5, 2)` due to the current rounding, and such rounding would not be required.

**Review by dualspiral — APPROVED** [review]
It does what it says on the tin, but there is a question here as to whether we could do something else and avoid square root calls.

We could pull as is if we think it's not worth doing that right now, or if you want to split that off into another PR.

File: src/main/java/org/spongepowered/common/mixin/plugin/tileentityactivation/TileEntityActivation.java (hunk 1 above)
Comment (ImMorpheus) [reply src/main/java/org/spongepowered/common/mixin/plugin/tileentityactivation/TileEntityActivation.java]: Good point.
I've switched to `Vector3i#distanceSquared` (which returns an int, so there's no need for rounding).

**Review by gabizou — APPROVED** [review]
*eventually* can have improved code, but this is improvement already.

**Pull request merged**